### PR TITLE
Use jsonrpc core types more

### PR DESCRIPTION
**Pull request by faern**
I realized that all request and response types are present in `jsonrpc-core` with correct serialize and deserialize implementations. So better to use them than this home built stuff.

The goal is to use it more on both the request side and the response side. But this PR starts off small with using it only in the response part.

---
This change is [Reviewable](https://reviewable.io/reviews/mullvad/jsonrpc-client-rs/27)

### CODE REVIEW

**Comment by faern** [comment]
Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion.

---

*[core/src/lib.rs, line 154 at r1](https://reviewable.io:443/reviews/mullvad/jsonrpc-client-rs/27#-KxbtlmsaaCA2lvTEVPZ:-KxbtlmsaaCA2lvTEVP_:brttqzw) ([raw file](https://github.com/mullvad/jsonrpc-client-rs/blob/1d588851b1aba6a6bc029c54765c09ad32d18eef/core/src/lib.rs#L154)):*
> ```Rust
>     T: Transport,
>     P: serde::Serialize,
>     R: serde::de::DeserializeOwned + Send + 'static,
> ```

So the `DeserializeOwned` trait is basically just an alias for `for<'de> serde::Deserialize<'de>`. So the bounds might look a bit simpler without specifying the lifetime etc.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/jsonrpc-client-rs/27)*

**Comment by faern** [comment]
I went ahead and pushed the improvement of the serialization as well.

File: core/src/lib.rs [hunk 1]
```diff
@@ -183,20 +190,26 @@ where
...
-    serde_json::to_vec(&JsonValue::Object(request_map))
+    let serialized_params = match serde_json::to_value(params)? {
+        JsonValue::Null => None,
+        JsonValue::Array(vec) => Some(Params::Array(vec)),
+        JsonValue::Object(obj) => Some(Params::Map(obj)),
+        value => Some(Params::Array(vec![value])),
+    };
+    let method_call = MethodCall {
+        jsonrpc: Some(Version::V2),
+        method: method.to_owned(),
```
Comment (tomusdrw) [inline core/src/lib.rs]: Seems to be owned already.

Comment (faern) [reply core/src/lib.rs]: Ah. Correct. A leftover from earlier when `method` was a `&str`. Fixing that.

**Comment by simonasker** [comment]
:lgtm:

---

Reviewed 3 of 3 files at r2.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/jsonrpc-client-rs/27#-:-KxfAZj2aWrjeSrIrBJs:bnfp4nl)*

**Pull request merged**